components/connectivity-adapter/internal/appregistry/service/converter.go

```diff
@@ -130,16 +134,29 @@ func (c *converter) DetailsToGraphQLCreateInput(deprecated model.ServiceDetails)
 		}
 
 		if deprecated.Api.Spec == nil { // TODO provide test for that
-			if deprecated.Api.SpecificationUrl != "" || deprecated.Api.SpecificationCredentials != nil || deprecated.Api.SpecificationRequestParameters != nil {
+
+			lowercaseDeprecatedAPIType := strings.ToLower(deprecated.Api.ApiType)
+			if deprecated.Api.SpecificationUrl != "" || deprecated.Api.SpecificationCredentials != nil || deprecated.Api.SpecificationRequestParameters != nil || lowercaseDeprecatedAPIType == oDataSpecType {
 				if apiDef.Spec == nil {
 					apiDef.Spec = &graphql.APISpecInput{}
 				}
-				apiDef.Spec.FetchRequest = &graphql.FetchRequestInput{
-					URL: deprecated.Api.SpecificationUrl,
-				}
 
 				apiDef.Spec.Type = toNewSpecType(deprecated.Api.ApiType)
-				apiDef.Spec.Format = graphql.SpecFormatJSON
+				if lowercaseDeprecatedAPIType == oDataSpecType {
+					apiDef.Spec.Format = graphql.SpecFormatXML
```

Comment:
I confirmed it and in Service Catalog UI, the documentation component supports only XML spec for OData. Newer OData APIs expose endpoint which can return XML or JSON spec.